httpbakery/agent/protocol.go

```diff
@@ -0,0 +1,76 @@
+package agent
+
+/*
+PROTOCOL
+
+The agent protocol is initiated when attempting to perform a
+discharge. It works as follows:
+
+A is Agent
+L is Login Service
+
+A->L
+	POST /discharge
+L->A
+	Interaction-required error containing
+	an entry for "agent" with field
+	"login-url" holding URL $loginURL.
+A->L
+	GET $loginURL?username=$user&public-key=$pubkey
+	where $user is the username to log in as
+	and $pubkey is the public key of that username,
+	base-64 encoded (standard encoding)
+L->A
+	JSON response:
+		macaroon:
+			macaroon with "local" third-party-caveat
+			addressed to $pubkey.
+A->L
+	POST /discharge?token=self-discharged macaroon
```

Comment:
Done, but does it make that much of a difference?  Macaroons need to be able to fit into headers anyway, and this one isn't likely to be that large (the one in the test came in at 332 bytes after URL encoding)